src/main/scala/chisel3/util/Arbiter.scala

```diff
@@ -18,6 +18,7 @@ class ArbiterIO[T <: Data](gen: T, n: Int) extends Bundle {
   val in  = Flipped(Vec(n, Decoupled(gen)))
   val out = Decoupled(gen)
   val chosen = Output(UInt(log2Ceil(n).W))
+  override def cloneType: this.type = new ArbiterIO(gen, n).asInstanceOf[this.type]
```

Comment:
I actually think we should change the constructor arguments to `private val gen: T, val n: Int` to take advantage of autoclonetype rather than implementing a clonetype.

I ran into an interesting issue trying to get all of this working in rocket-chip. If you override clonetype and then someone extends the class, you get runtime exceptions. This occurs in rocket-chip with `QueueIO` extended [here](https://github.com/freechipsproject/rocket-chip/blob/e7de7f3e823ffbf95fef2a0b9a6f143783393b84/src/main/scala/util/ShiftQueue.scala#L16).

I think we should change all of the library types to use autoclonetype